File: QadeerApp.Web/Modules/Administration/TrainingCalendarNote/TrainingCalendarNoteForm.cs
namespace QadeerApp.Administration.Forms;

[FormScript("Administration.TrainingCalendarNote")]
[BasedOnRow(typeof(TrainingCalendarNoteRow), CheckNames = true)]
public class TrainingCalendarNoteForm
{
    public string Title { get; set; }
    public string NoteText { get; set; }
    public int TrainingCalendarId { get; set; }
    public DateTime NoteDate { get; set; }
    public bool IsActive { get; set; }
}

File: QadeerApp.Web/Modules/ServerTypes/Administration/TrainingCalendarNoteForm.ts
import { BooleanEditor, DateEditor, initFormType, LookupEditor, PrefixedContext, StringEditor } from "@serenity-is/corelib";

export interface TrainingCalendarNoteForm {
    Title: StringEditor;
    NoteText: StringEditor;
    TrainingCalendarId: LookupEditor;
    NoteDate: DateEditor;
    IsActive: BooleanEditor;
}

export class TrainingCalendarNoteForm extends PrefixedContext {
    static readonly formKey = 'Administration.TrainingCalendarNote';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!TrainingCalendarNoteForm.init) {
            TrainingCalendarNoteForm.init = true;

            var w0 = StringEditor;
            var w1 = LookupEditor;
            var w2 = DateEditor;
            var w3 = BooleanEditor;

            initFormType(TrainingCalendarNoteForm, [
                'Title', w0,
                'NoteText', w0,
                'TrainingCalendarId', w1,
                'NoteDate', w2,
                'IsActive', w3
            ]);
        }
    }
}